charts/opentelemetry-collector/values.yaml

```diff
@@ -83,6 +83,9 @@ configMap:
   # Specifies an existing ConfigMap to be mounted to the pod
   # The ConfigMap MUST include the collector configuration via a key named 'relay' or the collector will not start.
   existingName: ""
+  # Must be specified when configMap.create = false to ensure deployment restart when the configMap changes, using
+  # a checksum.config annotation.
+  configTemplate: ""
```

Comment:
What does this new field do? In your previous commit it looked like we could get away with an If-Else statement. I was hoping we could move that logic into the helper.

Reply:
I was trying to be explicit since the name of the template file might not match the name of the ConfigMap in k8s.

But if we prefer to be implicit to reduce the amount of config, I'm okay with that. I'll try to re-use the `existingName` so we can see how that feels.

Reply:
Is [this](https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1302/commits/9d65a686207b9a2bf021a1ce689ff180d79c769f) closer to what we want?